src/js/api/errors.js

```diff
@@ -150,6 +155,22 @@ export const MSG_PROTECTED_CONTENT = 'protectedContent';
  */
 export const MSG_TECHNICAL_ERROR = 'technicalError';
 
+/**
+ * @enum {ErrorKey}
+ */
+export const MSG_PLUGIN_ERROR = 'pluginError';
+
+/**
+ * @enum {ErrorKey}
+ */
+export const MSG_CAPTIONS_ERROR = 'captionsError';
+
+/**
+ * @enum {ErrorKey}
+ */
+export const MSG_NETWORK_ERROR = 'networkError';
```

Comment:
I don't see the need for keys on warnings. The keys are used to apply a message  in `triggerError` and then removed:

```js
triggerError(evt) {
        const model = this._model;
        evt.message = model.get('localization').errors[evt.key];
        delete evt.key;
```
So adding keys to warnings just leaves us with a new key property not found in error events. Please remove.